compiler/rustc_data_structures/src/sync.rs

```diff
@@ -92,6 +93,23 @@ mod mode {
 
 pub use mode::{is_dyn_thread_safe, set_dyn_thread_safe_mode};
 
+fn catch<R>(
+    store: &Lock<Option<Box<dyn Any + std::marker::Send + 'static>>>,
```

Comment:
Why a lock, and not a simple `&mut`?

Reply:
It's intended to be used in parallel loops, so it wouldn't normally have mutable access.